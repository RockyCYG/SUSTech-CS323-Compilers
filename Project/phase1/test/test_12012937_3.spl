
struct class{
	int pair[90000];
	float k[0]=0x0;
	int _3333_=ic[kkk];
}
;char ccpc(int cy,char ababab){
	int cy = '@';
	char iii = '-';
	int vector = string;
	class.pa.x.yyy.t.tt.t.t.t.t.s = 233;
}
