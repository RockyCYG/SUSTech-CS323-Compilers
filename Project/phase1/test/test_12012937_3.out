Program (2)
  ExtDefList (2)
    ExtDef (2)
      Specifier (2)
        StructSpecifier (2)
          STRUCT
          ID: class
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    VarDec (3)
                      ID: pair
                    LB
                    INT: 90000
                    RB
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: float
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      VarDec (4)
                        ID: k
                      LB
                      INT: 0
                      RB
                    ASSIGN
                    Exp (4)
                      INT: 0
                SEMI
              DefList (5)
                Def (5)
                  Specifier (5)
                    TYPE: int
                  DecList (5)
                    Dec (5)
                      VarDec (5)
                        ID: _3333_
                      ASSIGN
                      Exp (5)
                        Exp (5)
                          ID: ic
                        LB
                        Exp (5)
                          ID: kkk
                        RB
                  SEMI
          RC
      SEMI
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: char
        FunDec (7)
          ID: ccpc
          LP
          VarList (7)
            ParamDec (7)
              Specifier (7)
                TYPE: int
              VarDec (7)
                ID: cy
            COMMA
            VarList (7)
              ParamDec (7)
                Specifier (7)
                  TYPE: char
                VarDec (7)
                  ID: ababab
          RP
        CompSt (7)
          LC
          DefList (8)
            Def (8)
              Specifier (8)
                TYPE: int
              DecList (8)
                Dec (8)
                  VarDec (8)
                    ID: cy
                  ASSIGN
                  Exp (8)
                    CHAR: '@'
              SEMI
            DefList (9)
              Def (9)
                Specifier (9)
                  TYPE: char
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: iii
                    ASSIGN
                    Exp (9)
                      CHAR: '-'
                SEMI
              DefList (10)
                Def (10)
                  Specifier (10)
                    TYPE: int
                  DecList (10)
                    Dec (10)
                      VarDec (10)
                        ID: vector
                      ASSIGN
                      Exp (10)
                        ID: string
                  SEMI
          StmtList (11)
            Stmt (11)
              Exp (11)
                Exp (11)
                  Exp (11)
                    Exp (11)
                      Exp (11)
                        Exp (11)
                          Exp (11)
                            Exp (11)
                              Exp (11)
                                Exp (11)
                                  Exp (11)
                                    Exp (11)
                                      ID: class
                                    DOT
                                    ID: pa
                                  DOT
                                  ID: x
                                DOT
                                ID: yyy
                              DOT
                              ID: t
                            DOT
                            ID: tt
                          DOT
                          ID: t
                        DOT
                        ID: t
                      DOT
                      ID: t
                    DOT
                    ID: t
                  DOT
                  ID: s
                ASSIGN
                Exp (11)
                  INT: 233
              SEMI
          RC
